pkg/analysis/passes/nestedmetadata/nestedmetadata.go

```diff
@@ -0,0 +1,108 @@
+package nestedmetadata
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/bmatcuk/doublestar/v4"
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/archive"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+)
+
+var (
+	missingMetadata      = &analysis.Rule{Name: "missing-metadata", Severity: analysis.Error}
+	errorReadingMetadata = &analysis.Rule{Name: "error-reading-metadata", Severity: analysis.Error}
+	invalidMetadata      = &analysis.Rule{Name: "invalid-metadata", Severity: analysis.Error}
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "nestedmetadata",
+	Requires: []*analysis.Analyzer{archive.Analyzer},
+	Run:      run,
+	Rules:    []*analysis.Rule{missingMetadata, errorReadingMetadata},
+}
+
+var MainPluginJson = "plugin.json"
+
+type Metadatamap map[string]metadata.Metadata
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	archiveDir, ok := pass.ResultOf[archive.Analyzer].(string)
+
+	if !ok {
+		return nil, nil
+	}
+
+	pluginJsonPath, err := doublestar.FilepathGlob(archiveDir + "/**/plugin.json")
```

Comment:
Good point. But the validator doesn't target native windows. There are plenty of other places in the code that won't work in windows because of similar path reasons.

Perhaps we should open an issue and fix them across the source? Most people are probably using the validator through the CI actions that run in linux runners. Though there's the NPX wrapper that people might run in a native windows.

Reply:
Sounds good :+1: